feat: add sitemap framework

Jer-Pha opened the pull request:
Fixes https://github.com/sandiegopython/pythonsd-django/issues/129

Jer-Pha committed:
feat: add sitemap framework

davidfischer commented on pythonsd/sitemap.py:
```diff
@@ -0,0 +1,34 @@
+class IndexSitemap(Sitemap):
+	changefreq = 'weekly'
+	protocol = 'https'
+	lastmod = datetime(2024, 5, 15, 0, 0, 0, tzinfo=timezone.utc)
```
I wonder if we should just remove this since it's going to quickly get out of date. It is [optional](https://docs.djangoproject.com/en/4.2/ref/contrib/sitemaps/#django.contrib.sitemaps.Sitemap.lastmod). Thoughts?

```suggestion
```

Jer-Pha committed:
Update pythonsd/sitemap.py

Agreed. I believe it might be slightly better for SEO during the initial crawl by Bing/Google search consoles, but it would need to be removed post-crawl anyway.

Co-authored-by: David Fischer <djfische@gmail.com>

davidfischer commented on pythonsd/sitemap.py:
```diff
@@ -0,0 +1,33 @@
+class StaticViewSitemap(Sitemap):
+	changefreq = 'monthly'
+	protocol = 'https'
+	lastmod = datetime(2024, 5, 15, 0, 0, 0, tzinfo=timezone.utc)
```
```suggestion
```

davidfischer committed:
Apply suggestions from code review

Jer-Pha committed:
removed unused imports

davidfischer commented:
This looks good and tested out. Thanks for the contribution.
